editor/src/components/editor/store/store-deep-equality-instances.ts

```diff
@@ -1035,6 +1038,8 @@ export function JSXElementChildKeepDeepEquality(): KeepDeepEqualityCall<JSXEleme
       return JSXTextBlockKeepDeepEquality(oldElement, newElement)
     } else if (isJSXFragment(oldElement) && isJSXFragment(newElement)) {
       return JSXFragmentKeepDeepEquality(oldElement, newElement)
+    } else if (isJSXConditionalExpression(oldElement) && isJSXConditionalExpression(newElement)) {
```

Comment:
Oooh, that's upsetting. Not the fix, but that this issue was in there.

Reply:
Yeah it's... puzzling, it'd be nice to have the compiler yell at us there

Reply:
We could change this to a switch/case statement, which would make this slightly better, just not as nice as we still have to check both types.